src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -656,16 +656,11 @@ private bool CheckFieldValueKind(SyntaxNode node, BoundFieldAccess fieldAccess,
         private static bool CheckSimpleAssignmentValueKind(SyntaxNode node, BoundAssignmentOperator assignment, BindValueKind valueKind, DiagnosticBag diagnostics)
         {
             // Only ref-assigns produce LValues
-            if (assignment.IsRef)
+            if (assignment.IsRef &&
+                (!RequiresAssignableVariable(valueKind) ||
```

Comment:
I would just call CheckValueKind on the left recursively. 

There are other value kinds such as "ref-returnable". Lets just defer all checks to the LHS. - for the same reason we would want to defer writeability.

Will need a test for "ref-returnable", I guess.

Reply:
But there's no recursion here, is there? A ref assignment must have a variable on the left hand, so this can't go any deeper than one level. This also may make error production worse because CheckValueKind would produce an error on the LHS of the assignment, even though there's nothing wrong with the LHS here, only the expression as a whole.

Reply:
I guess we are duplicating code here by checking the readonlyness twice. I'll call CheckValueKind.